core/models/identity.py

```diff
@@ -157,6 +157,9 @@ def create_identity(cls, username, provider_location,
             c_key = c_key.replace('cred_', '')
             credentials[c_key] = c_value
 
+        if 'router_name' not in credentials:
+            credentials['router_name'] = provider.select_router()
+
```

Comment:
I wouldn't find it intuitive (though I'm wrong, likely) that a provider would have a selected router based on a credential. 

Maybe I don't understand `ProviderCredential`?